lib/base-service.js

```diff
@@ -3,11 +3,20 @@ class BaseServiceSocket {
     this._socketClient = socketClient;
   }
 
+  _assignClientFailureHandlers (...sourceStreams) {
+    for (const evt in ['close', 'end']) {
+      this._socketClient.once(evt,
+        () => sourceStreams.map((s) => s.unpipe(this._socketClient)));
+    }
+  }
+
   /**
    * Closes the underlying socket communicating with the phone
    */
   close () {
-    this._socketClient.end();
+    if (!this._socketClient.destroyed) {
```

Comment:
nit: I usually prefer returning early if a close is no longer necessary

Reply:
with return guard it would look like
```
if (this._socketClient.destroyed) {
  return;
}
this._socketClient.end();
```

which is longer than the current implementation